From 20f103070dfb6e55b40e4cefb3c78a64ab047df2 Mon Sep 17 00:00:00 2001
From: Danh Duong <danhdc1@fsoft.com.vn>
Date: Thu, 4 Oct 2018 15:58:52 +0700
Subject: Workaround to reduce the maximum latency for pkill_thread_latency

Description:
    1. Set maximum priority for threads on RT system.
    2. For synchronization between 2 threads, sleepping 300us on sending threads
	   before starting to send signal.

Signed-off-by: Danh Duong <danhdc1@fsoft.com.vn>
---
 testcases/realtime/func/pthread_kill_latency/pthread_kill_latency.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/testcases/realtime/func/pthread_kill_latency/pthread_kill_latency.c b/testcases/realtime/func/pthread_kill_latency/pthread_kill_latency.c
index 08b6488..e88eae5 100644
--- a/testcases/realtime/func/pthread_kill_latency/pthread_kill_latency.c
+++ b/testcases/realtime/func/pthread_kill_latency/pthread_kill_latency.c
@@ -54,7 +54,7 @@
 #include <librttest.h>
 #include <libstats.h>

-#define PRIO 89
+#define PRIO 99
 #define ITERATIONS 10000
 #define HIST_BUCKETS 100
 #define THRESHOLD 20
@@ -255,6 +255,7 @@ void *signal_sending_thread(void *arg)
	for (i = 0; i < ITERATIONS; i++) {

		debug(DBG_DEBUG, "Sending signal. Loopcnt = %d\n", i);
+		usleep(300);

		/* Record the time just before sending the signal */
		begin = rt_gettime();
--
2.7.4
